src/test/javascript/spec/component/course/course-overview.component.spec.ts

```diff
@@ -645,4 +665,34 @@ describe('CourseOverviewComponent', () => {
 
         expect(component.dropdownOpen).toBeTrue();
     });
+
+    it('should initialize courses attribute when page is loaded', () => {
+        component.ngOnInit();
+
+        expect(component.courses).toEqual(courses);
+        expect(component.courses?.length).toBe(1);
+    });
+
+    it('should not initialize courses attribute when page has error while loading', () => {
+        findAllForDashboardSpy.mockReturnValue(throwError(() => new HttpResponse({ status: 404 })));
+
+        component.ngOnInit();
+        expect(component.courses?.length).toBeUndefined();
+    });
```

Comment:
it was rather written to keep the findAllForDashboardSpy as it throws a not used error otherwise eventough i do define the behaviour of it and use it. 😅